### PR TITLE
refactor controller manager: keep code style consistent among all controllers

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
1. remove unused code: include useless check for clientset and useless method
2. keep code style consistent among all controllers
3. replace deprecated interface with versioned ones

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/kube-controller-manager/app/core.go [hunk 1]
```diff
@@ -59,7 +59,7 @@ import (
 )
 
 func startServiceController(ctx ControllerContext) (bool, error) {
-	serviceController, err := servicecontroller.New(
+	serviceController, err := servicecontroller.NewServiceController(
```
Comment (janetkuo) [inline cmd/kube-controller-manager/app/core.go]: `servicecontroller.New` is preferred, see https://github.com/golang/go/wiki/CodeReviewComments#package-names

Comment (hzxuzhonghu) [reply cmd/kube-controller-manager/app/core.go]: i see, but also many other controllers use long names.

Comment (janetkuo) [reply cmd/kube-controller-manager/app/core.go]: I'd suggest either leave it as is or change other controllers' `New...` function names

Comment (hzxuzhonghu) [reply cmd/kube-controller-manager/app/core.go]: OK. I see

**Comment by hzxuzhonghu** [comment]
close this, for no great need

**Pull request closed without merging**